backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ConnectionManager.java

```diff
@@ -226,7 +237,11 @@ private void resolveAndSend(ConnectionRecord record, Partner p) {
         // only incoming connections in state request
         if (ConnectionState.REQUEST.equals(record.getState())) {
             didResolver.lookupIncoming(p);
-            messageService.sendMessage(WebSocketMessageBody.partnerReceived(conv.toAPIObject(p)));
+            if (record.isIncomingConnection()) {
```

Comment:
This ignores the auto flag and always fires the event, this is not what we want. The ConnectionRecord already has a method isRequestToConnect() that checks exactly that. The goal is to move all those checks into the event objects, because otherwise we will have them again and again all over the place, and as they can get pretty complex it easy to miss something.

Also this method is only called if incoming